compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -518,8 +519,18 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         &self,
         frame: &Frame<'mir, 'tcx, M::PointerTag, M::FrameExtra>,
         value: T,
-    ) -> T {
-        frame.instance.subst_mir_and_normalize_erasing_regions(*self.tcx, self.param_env, value)
+    ) -> Result<T, InterpError<'tcx>> {
+        frame
+            .instance
+            .try_subst_mir_and_normalize_erasing_regions(*self.tcx, self.param_env, value)
+            .or_else(|e| {
+                self.tcx.sess.delay_span_bug(
```

Comment:
Not sure whether it's correct to use a `delay_span_bug` here, or more specifically whether normalization failure during const evaluation would always result in an error.

Reply:
I don't know either, but if the test passes with the delay_span_bug then it seems reasonable to keep it until we find a testcase that says otherwise. ;)

Reply:
I can't come up with anything that would fail right now but not fail more loudly later. So yea, let's keep it in.